### PR TITLE
VM: Add disk I/O limit support - from Incus

### PR DESCRIPTION
**Pull request by simondeziel**
I feel this warrants an API extension but none was added in Incus.

### CODE REVIEW

**Comment by tomponline** [comment]
> I feel this warrants an API extension but none was added in Incus.

Agreed.

It would also be useful to add some tests to lxd-ci for this if possible?

**Comment by simondeziel** [comment]
> > I feel this warrants an API extension but none was added in Incus.
> 
> Agreed.

So I should add the extension and collapse all those commits into one, with proper attribution, right?

> It would also be useful to add some tests to lxd-ci for this if possible?

That's indeed the plan.

**Comment by tomponline** [comment]
> So I should add the extension and collapse all those commits into one, with proper attribution, right?

No, just add a single api extension commit (see https://github.com/canonical/lxd/blob/main/CONTRIBUTING.md#commit-structure) and that can be attributed to yourself as normal.

**Comment by tomponline** [comment]
It doesn't look like there are any doc commits either so it'll also need something updated in the docs to indicate those keys are suitable for VMs and what they work on. cc @ru-fu

File: doc/api-extensions.md [hunk 1]
```diff
@@ -2363,7 +2363,6 @@ This API extension indicates that the `/1.0/instances/{name}/uefi-vars` endpoint
 This API extension allows newly created VMs to have their `migration.stateful` configuration key automatically set
 through the new server-level configuration key `instances.migration.stateful`. If `migration.stateful` is already set at the profile or instance level then `instances.migration.stateful` is not applied.
 
-## `access_management`
+## `vm_disk_io_limits`
```
Comment (ru-fu) [inline doc/api-extensions.md:2366]: Is it intentional that you removed access_management?

Comment (simondeziel) [reply doc/api-extensions.md]: Definitely not, thanks for spotting. Fixed now.

**Comment by simondeziel** [comment]
@tomponline the API extension was added, please take another look.

**Review by tomponline — APPROVED** [review]
@simondeziel have you tested manually that this does work btw?

**Comment by simondeziel** [comment]
@tomponline I tested just `limits.read=1000` which causes IOs to plummet down to reading at ~1Kbps:

```
$ lxc exec v1 -- dd if=/dev/sda15 of=/dev/null iflag=direct status=progress
106263040 bytes (106 MB, 101 MiB) copied, 12 s, 8.9 MB/s
217088+0 records in
217088+0 records out
111149056 bytes (111 MB, 106 MiB) copied, 12.5408 s, 8.9 MB/s
$ lxc config device set v1 root limits.read=1000
$ lxc exec v1 -- dd if=/dev/sda15 of=/dev/null iflag=direct status=progress
17920 bytes (18 kB, 18 KiB) copied, 17 s, 1.0 kB/s^C
37+0 records in
36+0 records out
18432 bytes (18 kB, 18 KiB) copied, 18.3342 s, 1.0 kB/s
```

**Comment by tomponline** [comment]
Thanks!

**Comment by tomponline** [comment]
> @tomponline I tested just `limits.read=1000` which causes IOs to plummet down to reading at ~1kBps:
> 
> ```
> $ lxc exec v1 -- dd if=/dev/sda15 of=/dev/null iflag=direct status=progress
> 106263040 bytes (106 MB, 101 MiB) copied, 12 s, 8.9 MB/s
> 217088+0 records in
> 217088+0 records out
> 111149056 bytes (111 MB, 106 MiB) copied, 12.5408 s, 8.9 MB/s
> $ lxc config device set v1 root limits.read=1000
> $ lxc exec v1 -- dd if=/dev/sda15 of=/dev/null iflag=direct status=progress
> 17920 bytes (18 kB, 18 KiB) copied, 17 s, 1.0 kB/s^C
> 37+0 records in
> 36+0 records out
> 18432 bytes (18 kB, 18 KiB) copied, 18.3342 s, 1.0 kB/s
> ```

Sounds like a something maybe we can test on lxd-ci?

**Pull request merged**